apps/api/src/app/integrations/usecases/check-integration/check-integration.usecase.ts

```diff
@@ -8,9 +8,19 @@ export class CheckIntegration {
   constructor(private checkIntegrationEmail: CheckIntegrationEMail) {}
 
   public async execute(command: CheckIntegrationCommand) {
-    switch (command.channel) {
-      case ChannelTypeEnum.EMAIL:
-        return await this.checkIntegrationEmail.execute(command);
+    try {
+      switch (command.channel) {
+        case ChannelTypeEnum.EMAIL:
+          return await this.checkIntegrationEmail.execute(command);
+      }
+    } catch (e) {
+      if (e.message?.includes('getaddrinfo ENOTFOUND')) {
+        throw new NotFoundException(
+          `Provider gateway can't resolve the host with the given hostname ${command.credentials?.host || ''}`
+        );
+      }
```

Comment:
Added additional catch so we could provide more information error to the user. i was not sure what status to add because this one is not found gateway (502) but not in novu platform.

Reply:
If I would change the host to something correct I won't get this error, right? if yes, I feel it should be more bad request error